tests/unit/conftest.py

```diff
@@ -49,12 +50,14 @@ def email_app():
         CELERY_EAGER_PROPAGATES_EXCEPTIONS=True,
         MAIL_SUPPRESS_SEND=True,
     )
+    app.config.update(PRESERVE_CONTEXT_ON_EXCEPTION=False,)
```

Comment:
I'll try removing this too, this is supposed to fix the 'popping wrong app context' issue, but it did not, the new code does not have that issue though, so it might not be needed